### DISCUSSION TITLE
AWA protocol till 1,5MB No more?

### DISCUSSION THREAD

**Discussion by wou5er** [awawa-dev/HyperHDR#250] [Q&A]
Hello,

I'm using very satisfied your project, but i'm only able to use it at 1,5 Mb using HyperserialWLED.

As i took a look at the chip there isn't any info on it so my bet is its a cp chip
Fixing a 340 chip ;solves the issue?

**Comment by awawa-dev** [comment]
Hi
Are you using some old betas or latest stable release of HyperSerialWLED (based on 0.13.1): ? https://github.com/awawa-dev/HyperSerialWLED/releases/tag/v6.0.0.0
I ask because WLED offers only 1.5Mb in the selection list and I extended it on my own (to 2Mb and 4Mb) in the latest release.
If the latest WLED is used and you've got 2Mb speed but it doesn't work then it's a chip problem. It's hard to find genuine esp8266 ch340c/g lately and perhaps ESp32 CH9102x is a better choice (btw 4Mb capable). So far I haven't heart about counterfeited CH9102x . Also with ESP32 we're avoiding all the mess with WLED pinout remapping...

**Comment by wou5er** [comment]
I'm using latest release v6.0.0.0 cause i've got the options 2 and 4mb but they both don't work.
Thanks for confirm my thoughts and for the tip, I,ll have look for the CH9102x
But i don't have any troubles with remapping using ws2801 

found [this](https://www.banggood.com/ESP-32S-CH9102X-QFN28-ESP32-Development-Board-Wireless-WiFi-+-Bluetooth-2-in-1-Dual-core-CPU-Low-Power-Consumption-ESP-32-Control-Board-p-1883633.html?rmmds=myorder&cur_warehouse=CN) one on wich pins does my clock and data goes

**Comment by awawa-dev** [comment]
Well, you are lucky and ws2801 is low density strip but for typical APA102 setup it's causing massive headache if not SPI hardware pins are selected on esp8266. Or you set your smoothing conservative, because above 30FPS WLED (in normal circumstance with correct setup 100-200FPS is possible) may become unresponsive.

**Reply by wou5er** [reply]
As till this point i'm able to use 100hz and 200hz @ 1,5Mb @ 60 fps using a EZcap 321 wich runs perfect.. 

Orded the ESP now we'll wait... 

My ambilight hasn't been that great since i ever started  with this 3 years ago, very nice work

**Comment by wou5er** [comment]
So i got my ESP today and started playing with it... Final thoughts... things arent so easy to port to the NodeMCU. 
I made a exact copy of the wiring with a Level shifter and even moved the wire combination from my earlier Wemos D1 clone to the NodeMCU but some how i'm not able to get it stop flickering. 
Am i missing something?  Wrong Level Shifter read about it in the past those things can be tricky... but then again this one should be supported.
ESP, LED and Pi have a Common Ground.

**Reply by awawa-dev** [reply]
Maybe it's a WLED configuration issue (power limit etc.)? If you are using AWA protocol then data received by ESP are 100% valid so it must be something else. Sometimes too long cable/path for the common ground (or with some loops or bridges) may cause such issue.

**Reply by wou5er** [reply]
Could be my Clock and Data wires aren't the shortes... over 10cm xD common grounds are also not that tight to each other.. 
But then again why is this not working on a 107 LEDs config.. i did get it working on only 36 with no problems..

Awa protocol 4mb and lower works but it flickers a lot so i think i can go with your thaughts about hardware config

**Reply by awawa-dev** [reply]
You've got quite long story about how wrong common ground path may cause exactly such symptoms and it's not about its length: https://github.com/awawa-dev/HyperHDR/discussions/191 
If it depends on the LED number then probably something is wrong with WLED: it's a common issue but only for ESP8266 and SPI LED strips. Try alternative driver.

**Reply by wou5er** [reply]
> 

To wich part are you refering? there's a lot of info.

this part?:

Your diagram looks good. You need:

> flash ESP32(with CP2104/CH9102x on board) with HyperSerialESP32 (for example: RGBW cold white version firmware). On Windows I'm using esphome-flasher
> connect Mac Mini with ESP32 using USB cable
> connect ESP32 data line output (output pin is described on HyperSerialESP32 page) with level shifters input. The connection diagram in previous post.
> connect data line from the level shifter output to sk6812 data line input.
> You need also standard "common ground": the ground of the ESP32 (from its pinout) should be connected to the "minus" from the LED strip power supply, that is also connected to the "minus" of the LED strip. Without it very bad things could happen.

**Reply by awawa-dev** [reply]
How the common ground  was routed and was causing flickering. Non standard problem so a lot of reading but most interesting is near the end. But if I had to guess your problem is probably WLED itself.

**Reply by awawa-dev** [reply]
https://github.com/awawa-dev/HyperHDR/discussions/191#discussioncomment-2162246

**Reply by wou5er** [reply]
Funny enough I got that exact same thing. 
BUT
only happens when using it on the Nodemcu on my 107 LED setup.

Mayby try the offline version without wled?

**Reply by awawa-dev** [reply]
You can use HyperSerial instead of WLED and see if the problems still exists.

**Comment by wou5er** [comment]
So having Hyperserial flashed... but HyperHDR isn't sending any signal. got no response on my leds and in hyperhdr cannot change source under remote control.
Am I missing something?

Btw i power esp32 with external power supply and connected with usb to pi without the power line.

here is my log:

```
2022-04-20T12:40:54.556Z [LEDDEVICE] Start LedDevice 'adalight'.
2022-04-20T12:40:54.557Z [LEDDEVICE] (LedDevice.cpp:149) deviceConfig: [{"awa_mode":true,"colorOrder":"rgb","currentLedCount":107,"delayAfterConnect":0,"hardwareLedCount":107,"latchTime":15,"output":"auto","rate":115200,"rewriteTime":1000,"type":"adalight"}]
2022-04-20T12:40:54.557Z [LEDDEVICE] (LedDevice.cpp:408) LatchTime updated to 15ms
2022-04-20T12:40:54.558Z [LEDDEVICE] (LedDevice.cpp:429) Refresh interval = 1000ms
2022-04-20T12:40:54.558Z [LEDDEVICE] (LedDevice.cpp:435) RewriteTime updated to 1000ms
2022-04-20T12:40:54.558Z [LEDDEVICE] (ProviderRs232.cpp:36) DeviceType   : adalight
2022-04-20T12:40:54.559Z [LEDDEVICE] (ProviderRs232.cpp:37) LedCount     : 107
2022-04-20T12:40:54.559Z [LEDDEVICE] (ProviderRs232.cpp:38) ColorOrder   : rgb
2022-04-20T12:40:54.559Z [LEDDEVICE] (ProviderRs232.cpp:39) RefreshTime  : 1000
2022-04-20T12:40:54.560Z [LEDDEVICE] (ProviderRs232.cpp:40) LatchTime    : 15
2022-04-20T12:40:54.560Z [LEDDEVICE] (ProviderRs232.cpp:52) deviceName   : auto
2022-04-20T12:40:54.560Z [LEDDEVICE] (ProviderRs232.cpp:53) AutoDevice   : 1
2022-04-20T12:40:54.561Z [LEDDEVICE] (ProviderRs232.cpp:54) baudRate_Hz  : 115200
2022-04-20T12:40:54.561Z [LEDDEVICE] (ProviderRs232.cpp:55) delayAfCon ms: 0
2022-04-20T12:40:54.562Z [LEDDEVICE] (LedDeviceAdalight.cpp:64) Adalight driver with activated high speeed & data integration check AWA protocol
2022-04-20T12:40:54.562Z [LEDDEVICE] (LedDeviceAdalight.cpp:74) Adalight header for 107 leds: Awa 0x00 0x6a 0x3f
2022-04-20T12:40:54.698Z [LEDDEVICE] found serial device: ttyACM0
2022-04-20T12:40:54.698Z [LEDDEVICE] Opening UART: ttyACM0
2022-04-20T12:40:54.699Z [LEDDEVICE] (ProviderRs232.cpp:141) _rs232Port.open(QIODevice::ReadWrite): ttyACM0, Baud rate [115200]bps
2022-04-20T12:40:54.839Z [LEDDEVICE] (ProviderRs232.cpp:146) portName:          ttyACM0
2022-04-20T12:40:54.841Z [LEDDEVICE] (ProviderRs232.cpp:147) systemLocation:    /dev/ttyACM0
2022-04-20T12:40:54.842Z [LEDDEVICE] (ProviderRs232.cpp:148) description:       USB Single Serial
2022-04-20T12:40:54.842Z [LEDDEVICE] (ProviderRs232.cpp:149) manufacturer:      1a86
2022-04-20T12:40:54.843Z [LEDDEVICE] (ProviderRs232.cpp:150) productIdentifier: 0x55d4
2022-04-20T12:40:54.844Z [LEDDEVICE] (ProviderRs232.cpp:151) vendorIdentifier:  0x1a86
2022-04-20T12:40:54.845Z [LEDDEVICE] (ProviderRs232.cpp:152) serialNumber:      5299007073
2022-04-20T12:40:54.850Z [COMPONENTREG0] LED device: enabled
2022-04-20T12:40:54.852Z [SMOOTHING0] Using linear smoothing input (0)
2022-04-20T12:40:54.853Z [SMOOTHING0] Enabling timer. Now timer is active: 1, remaining time to run: 5
2022-04-20T12:40:54.858Z [SMOOTHING0] Using linear smoothing procedure (0)
2022-04-20T12:40:57.455Z [WEBSOCKET] (JsonAPI.cpp:1300) log streaming activated for client ::ffff:192.168.2.200
```

**Reply by awawa-dev** [reply]
> "rate":115200

it's invalid for any HyperSerial (and my fork of WLED at default settings also)

**Reply by wou5er** [reply]
sorry for that changed it to 2.000.000 but still nothing happens
everythings looks okay i believe.?
```

2022-04-20T13:19:43.443Z [LEDDEVICE] Opening UART: ttyACM0
2022-04-20T13:19:43.443Z [LEDDEVICE] (ProviderRs232.cpp:141) _rs232Port.open(QIODevice::ReadWrite): ttyACM0, Baud rate [2000000]bps
2022-04-20T13:19:43.595Z [LEDDEVICE] (ProviderRs232.cpp:146) portName:          ttyACM0
2022-04-20T13:19:43.596Z [LEDDEVICE] (ProviderRs232.cpp:147) systemLocation:    /dev/ttyACM0
2022-04-20T13:19:43.596Z [LEDDEVICE] (ProviderRs232.cpp:148) description:       USB Single Serial
2022-04-20T13:19:43.596Z [LEDDEVICE] (ProviderRs232.cpp:149) manufacturer:      1a86
2022-04-20T13:19:43.597Z [LEDDEVICE] (ProviderRs232.cpp:150) productIdentifier: 0x55d4
2022-04-20T13:19:43.597Z [LEDDEVICE] (ProviderRs232.cpp:151) vendorIdentifier:  0x1a86
2022-04-20T13:19:43.597Z [LEDDEVICE] (ProviderRs232.cpp:152) serialNumber:      5299007073
2022-04-20T13:19:43.600Z [COMPONENTREG0] LED device: enabled
2022-04-20T13:19:45.889Z [WEBSOCKET] (Json
```API.cpp:1300) log streaming activated for client ::ffff:192.168.2.200

**Reply by awawa-dev** [reply]
Then you must proceed standard procedure to check if your esp in fact can communicate at 2Mb and use screen (```screen /dev/ttyAMA0 2000000```) to verify it. Important: hyperhdr process must be running first and trying to use LED and then it must be killed before you run screen to see the statistics.

**Reply by wou5er** [reply]
> 

how do i run that command "screen /dev/ttyAMA0 2000000"?

HyperHDR runs on background

**Reply by awawa-dev** [reply]
using ssh connection

**Reply by awawa-dev** [reply]
sorry, if it's too complicated I can't help with Linux basics further

**Reply by wou5er** [reply]
> using ssh connection

Yes i know that XD got some knowledge but not on how to run such a command when doing so i it gives bash command not found so it needs to run from a source wich is hyperhdr but where can i find that source?

**Reply by awawa-dev** [reply]
probably WLED is only applicable solution then, if it receives data via UDP/Wifi and it still flickering the problem is somewhere outside.

**Reply by awawa-dev** [reply]
google takes it under 30 seconds sudo apt-get install screen

**Reply by wou5er** [reply]
got no problem with basic linux just having issues not knowing were to find the place where to run this command?  any directions?

**Reply by wou5er** [reply]
> google takes it under 30 seconds sudo apt-get install screen

yeah sorry about that... i could do that myself  -.-"

what do i need to expect after running "screen /dev/ttyAMA0 2000000" 

What I do start HyperHDR, stop the service , run the command. SSH screen gets flushed and shows "screen is terminating"

**Reply by awawa-dev** [reply]
It should not behave like that unless hyperhdr process is still running and is using that serial port at /dev/ttyAMA0 so other applications including screen can't use it.

**Reply by wou5er** [reply]
Yeah i'll clean my setup in the future to clear everything out.... at the moment it works with the wemos.. happy for now mayby after sorting everything out it wil work agian.. 

as it works on only 36 leds on my desk... so proberly something isn't completly right behind the tv.

**Comment by wou5er** [comment]
Very weird problem here though. Did some troubleshooting.

Conclusion i don't know whats going on. 
I do know that the ESP is not working as expected because it does not work even on my test setup wich has over 63 LEDS in 2 strokes and even they are flickering
Common Ground shoudln't be a issue as they are both <10cm

**Reply by awawa-dev** [reply]
Did you try to run in in separated mode: completely no wire connection to Rpi, only Wifi data connection from Rpi? So separated LED power supply, ESP(WLED) and the LED strip?

**Reply by wou5er** [reply]
I did yes. On my Desk only 1 power supply to power esp and led strip with a common ground

Mayby i would think it could my level shifter but this is one wich should be supported but it has no ground on wich some level shifters do....

**Reply by awawa-dev** [reply]
why?  
> Btw i power esp32 with external power supply and connected with usb to pi without the power line.  

esp with HyperSerial (no Wifi) has very low power consumption. Completely no reason to feed it externaly. Rpi USB power supply is also stabilized just like ESP USB power in. Otherwise you must know what you are doing.

**Reply by awawa-dev** [reply]
and which level shifter you are using?

**Reply by wou5er** [reply]
Because feeding it directly made my pi instable... it has enough power to run the EZcap321 and i don't have a external powered usb hub by hand

**Reply by wou5er** [reply]
> and which level shifter you are using?

Pololu 2595

**Reply by awawa-dev** [reply]
> Because feeding it directly made my pi instable... it has enough power to run the EZcap321 and i don't have a external powered usb hub by hand  

No way it can destabilize Rpi if you provide just standard and proper power supply for Rpi. HyperSerial takes well below 100mA. Unless there is a bridge of +5V lines between two power suppliers which should not happen. Active USB HUB doesnt matter here. 

> Pololu 2595  

It's too slow, unbuffered and it's not recommended here. Search the HyperHDR wiki or discussions . Maybe if you are lucky it will work with neopixel led strips which are around 800Kb but your APA102 is around 5Mb in WLED. Generally we replace one problem (3.3V to 5V) with another (signal disruption).

**Reply by wou5er** [reply]
Would this level shifter be fast enough?

https://www.sparkfun.com/products/12009

I have this laying around?

**Reply by awawa-dev** [reply]
Don't know but it looks like mosfet type. Did you try without a such lever shifter because if it works then it make no sense to make it worse with uncertain level shifters? Comments in the shop are not optimistic.

**Reply by wou5er** [reply]
Back in the days I started with Hyperion i didn’t used a level shifter… this gave me problems… a shifter solved it but in those days my leds were directly connected to the pi over spi outputs

since then I never went without a shifter

**Reply by awawa-dev** [reply]
I guess you had lower speed set for SPI on Rpi. WLED and my ESP software are using higher speeds compared to typical RPI setups and probably it's too much for that level shifter. I have no other ideas.

**Comment by wou5er** [comment]
After all got a myself some 74AHCT125 level shifters and made a build works... but only till 2 MB

**Comment by awawa-dev** [comment]
OK, so ESP->LED is working and no blinking? Then I guess 2 MB limit hits the serial communication between WLED and RPI via USB? First you must examine if you have CH9102X or CP2104 because often sellers of ESP32 boards they don't know what version they've got on the stack and these two chips are often mixed up, they look similar and have similar design. CP2104 is capable of 2Mb and is being replaced by cheaper and more capable CH9102X  with almost no change to the ESP32 board design.

**Reply by wou5er** [reply]
Yes WLED and Pi via USB serial communication. 

I made a picture of the chip, I don't think its a CP2104  but mayby you can confirm?

https://ibb.co/bNL377X

**Comment by awawa-dev** [comment]
Yes, it's CH9102X and it looks like mine. And it's should work at 4Mb: confirmed in the spec and in the practice.

**Reply by wou5er** [reply]
I could try to power the esp trough Micro usb.. instead external… got one led connected to the esp.
It would be sufficient…

Ill do that tomorrow 😁 at some point it will work

**Reply by wou5er** [reply]
So i powering the ESP directly from the Pi but i'm still not able to hit over 4MB 

mayby a fake esp? i don't know ordered another one on ali... they state its a v2 with the new chip... 
the esp wich i already have i bought it here in the netherlands... i guess its still an old chip version.

**Reply by awawa-dev** [reply]
Don't have any previous reports about fakes of CH9102X and all of my few ESP32 boards equipped with that chip (looks exactly like on the picture) passed the tests.  And for comparison counterfeited ch340c/g have removed marking from the chips. Further diagnostic is possible but require programmer skills to exclude some configuration issues. Some difference in ESP board design or quality may or may have not some impact.

**Comment by wou5er** [comment]
Wich SBC you use for running HyperHDR on ? As of now I use a Raspberry Pi 4 wich does its job... but i'm thinking to upgrade te get a smoother capture experience and therefore a better ambilight experience . 
Was thinking of a Odroid H2 or just a cheap x86 cpu onboard kit.

**Comment by awawa-dev** [comment]
Rpi4 is fine. Especially if you are using the latest version of v18beta: it comes with some improvements and Rpi is able to process 1080p120 stream without any size/framerate decimation (v17 can't handle that). Well, nobody is using 1080p120 for our purposes, but it shows that there are tons of free resources for regular video processing. My experience with different TVs has taught me that a USB3.0 efficient driver and a stable operating system are an extremely important thing. And Rpi4 has both of them.

**Reply by wou5er** [reply]
Were do I find v18beta?

**Reply by wou5er** [reply]
I Need to compile to get V18?

**Reply by awawa-dev** [reply]
You can compile it on your own or you can download latest installer of beta version from HyperHDR github action tab.
![obraz](https://user-images.githubusercontent.com/69086569/165349959-2f8763f6-d0c1-4d6e-9538-edc83593409f.png)

**Reply by wou5er** [reply]
thank god found it. but after update/install hyperhdr v17 to v18. V18 does not start.. did an update and an clean install. am i missing something?

**Reply by awawa-dev** [reply]
run it from command line: to check out possible problems /usr/bin/hyperhdr 
if it works then the service is not starting from some reason
